py/mpconfig.h

```diff
@@ -731,6 +731,11 @@ typedef double mp_float_t;
 #define MICROPY_CAN_OVERRIDE_BUILTINS (0)
 #endif
 
+// Allow ports to provide a lazy load function for undefined globals
```

Comment:
Should state which function I guess, otherwise it's rather vague. Alternatively: make this macro be the name of the function e.g. port uses `#define MICROPY_LAZY_LOAD_GLOBAL my_lazy_load_global`.

Can you add an example of how this would be useful and why it's preferred over the port injecting globals in it's initialization code?

Reply:
> Should state which function I guess, otherwise it's rather vague. Alternatively: make this macro be the name of the function e.g. port uses #define MICROPY_LAZY_LOAD_GLOBAL my_lazy_load_global.

I'll state the name of the global. It's better this way because some header under `py/` will have the declaration of this function (so unlike an unknown function defined by the user, it's bound to have a specific prototype).

> Can you add an example of how this would be useful and why it's preferred over the port injecting globals in it's initialization code?

My intention really was to do it lazily, since there are possibly hundreds of thousands of symbols and I don't want to load all of them (I guess performance will be hit if the `globals()` dict will have 200k entries...). Instead, loading each global the user requested *once* is fair.

Reply:
I don't know what you mean with 'symbol' exactly, but is it really the best idea to allow having hundreds/thousands of those in global scope, where the sheer amount of them increases the likelyhood of name clashes with builtins/other items in that scope? I.e. can you provide a usage example, and would it be an option to do all of that in a separate module instead? Are all symbol names guranteed to be valid python identifiers?

Reply:
Moved this commit to https://github.com/micropython/micropython/pull/5522. I'm closing the discussion here, I mentioned this comment from the new PR, we'll continue it over there.